pkg/kubelet/network/plugins.go

```diff
@@ -199,3 +199,31 @@ func (plugin *NoopNetworkPlugin) GetPodNetworkStatus(namespace string, name stri
 func (plugin *NoopNetworkPlugin) Status() error {
 	return nil
 }
+
+// GetPodIP gets the IP of the pod by inspecting the network info inside the pod's network namespace.
+func GetPodIP(execer utilexec.Interface, id kubecontainer.ContainerID, nsenterPath, netnsPath, interfaceName string) (net.IP, error) {
```

Comment:
@freehan I didn't do this because the kubenet driver code caches the nsenterPath variable, but the cni driver code doesn't.
Ideally we should cache the nsenterPath in both drivers, but that will make the PR slightly bigger by adding mutex. Does that sound good to you?